src/Analyser/ResultCache/ResultCacheManager.php

```diff
@@ -1036,4 +1044,27 @@ private function getStubFiles(): array
 		return $stubFiles;
 	}
 
+	/**
+	 * @return array<string, string>
+	 * @throws ShouldNotHappenException
+	 */
+	private function getMetaFromPhpStanExtensions(): array
+	{
+		$meta = [];
+
+		/** @var ResultCacheMetaExtension $extension */
+		foreach ($this->container->getServicesByTag(ResultCacheMetaExtension::EXTENSION_TAG) as $extension) {
+			if (array_key_exists($extension->getKey(), $meta)) {
+				throw new ShouldNotHappenException(sprintf(
+					'Duplicate ResultCacheMetaExtension with key "%s" found.',
+					$extension->getKey(),
+				));
+			}
+
+			$meta[$extension->getKey()] = $extension->getHash();
+		}
+
+		return $meta;
```

Comment:
I just realized one more thing - let's sort the keys.

Reply:
Done.